beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepository.java

```diff
@@ -107,8 +109,37 @@ Long countRecordsForDryRunWherePartitionIsNotNullOrExpired(
       @Param("databaseName") String databaseName,
       @Param("tableName") String tableName);
 
+  /**
+   * This method deletes the rows for scheduled or failed partitions for the specified {@code databaseName} and
+   * {@code tableName}.
+   *
+   * @param databaseName
+   * @param tableName
+   */
+  @Modifying
+  @Query(value = "delete from HousekeepingMetadata t "
+      + "where t.databaseName = :databaseName "
+      + "and t.tableName = :tableName "
+      + "and t.partitionName is not NULL "
+      + "and (t.housekeepingStatus = 'SCHEDULED' or t.housekeepingStatus = 'FAILED')")
+  void deleteScheduledOrFailedPartitionRecordsForTable(@Param("databaseName") String databaseName,
+      @Param("tableName") String tableName);
+
+  /**
+   * This method returns all table records where the partition name is NULL and the status is `SCHEDULED` or `FAILED`
+   */
+  @Query(value = "from HousekeepingMetadata t "
+      + "where t.partitionName is NULL "
```

Comment:
oh or is this guaranteeing uniqueness as we only get the table entries of which we will only have 1?

Reply:
yes, exactly